server/src/project_service.ts

```diff
@@ -0,0 +1,142 @@
+import * as ts from 'typescript/lib/tsserverlibrary'; // used as value
+import * as lsp from 'vscode-languageserver';
+import {tsDiagnosticToLspDiagnostic} from './diagnostic';
+import {filePathToUri} from './utils';
+
+// NOTE:
+// There are three types of `project`:
+// 1. Configured project - basically all source files that belong to a tsconfig
+// 2. Inferred project - other files that do not belong to a tsconfig
+// 3. External project - not used in this context
+// For more info, see link below.
+// https://github.com/Microsoft/TypeScript/wiki/Standalone-Server-%28tsserver%29#project-system
+
+/**
+ * `ProjectService` is a singleton service for the entire lifespan of the
+ * language server. This specific implementation is a very thin wrapper
+ * around TypeScript's `ProjectService`. On creation, it spins up tsserver and
+ * loads `@angular/language-service` as a tsserver plugin.
+ * Care must be taken to load `@angular/language-service` from the project or
+ * workspace. This extension shoud not bundle its own
+ * `@angular/language-service`, not even as a backup. This is to ensure that
+ * the extension becomes a no-op when non-Angular TS projects are opened.
+ * `ProjectService` is used to manage both TS document as well as HTML.
+ * Using tsserver to handle non-TS files is fine as long as the ScriptKind is
+ * configured correctly and `getSourceFile()` is never called on non-TS files.
+ */
+export class ProjectService {
+  public readonly tsProjSvc: ts.server.ProjectService;
+
+  constructor(
+      private readonly serverHost: ts.server.ServerHost,
+      private readonly logger: ts.server.Logger,
+      private readonly connection: lsp.IConnection,
+      options: Map<string, string>,
+  ) {
+    const pluginProbeLocation =
+        options.get('pluginProbeLocation') || serverHost.getCurrentDirectory();
+    connection.console.info(
+        `Angular LS probe location: ${pluginProbeLocation}`);
+    // TODO: Should load TypeScript from workspace.
+    this.tsProjSvc = new ts.server.ProjectService({
+      host: serverHost,
+      logger,
+      cancellationToken: ts.server.nullCancellationToken,
+      useSingleInferredProject: true,
+      useInferredProjectPerProjectRoot: true,
+      typingsInstaller: ts.server.nullTypingsInstaller,
+      suppressDiagnosticEvents: false,
+      eventHandler: (e) => this.handleProjectServiceEvent(e),
+      globalPlugins: ['@angular/language-service'],
+      pluginProbeLocations: [pluginProbeLocation],
+      allowLocalPluginLoads: false,  // do not load plugins from tsconfig.json
+    });
+
+    this.tsProjSvc.configurePlugin({
+      pluginName: '@angular/language-service',
+      configuration: {
+        'angularOnly': true,
```

Comment:
This still provides TypeScript functionality, right?

Reply:
No, it does not provide TypeScript functionality. Please see comment above.